DT-6175 part 3 - more refactoring

vesameskanen opened the pull request.

vesameskanen committed:
fix: control panel component prop types

vesameskanen committed:
fix: move some functions to itinerary page utils, fix type errors, refactor

vesameskanen committed:
chore: rename utility file

vesameskanen committed:
fix: refactor bike & transit related props for itinerary list

- do not pass redundant props
- do not refilter already filtered itineraries
- refactor

vesameskanen committed:
fix: do not repeat identical logic for two street mode buttons

vesameskanen committed:
fix: refactor SteetModeSelector props and names

vesameskanen committed:
feat: new generic itinerary filtering func

vesameskanen committed:
fix: refactor itinerary management

- Filter once after fetching data, not at every access
- Merge bike + public itineraries into single array
- Reduce endless and confusing use of derived state variables

vesameskanen committed:
chore: update tests

vesameskanen committed:
chore: refactor legutils
- export named functions
- add utility function to extract leg mode

vesameskanen committed:
fix: use boolean type for conditional jsx rendering

vesameskanen committed:
fix: use simple code to find transit mode

vesameskanen committed:
fix: regression in showing car itineraries which are only for emissions

Also, fix some prop type errors, and simplify state variable use

vesameskanen committed:
chore: rename some variables which describe itineraty page state

optionsome reviewed (changes requested):
There is a conflict.

Pure bicycle itineraries don't show up anymore.

with HSL config (cars enabled) the following itinerary search crashed: `/reitti/Valittu sijainti%3A%3A60.350679325290315%2C24.472990036010742/Valittu sijainti%3A%3A60.16536744905669%2C24.84404146671295?time=1705615030` with stack trace:
<details>
Uncaught TypeError: leg is undefined
    getExtendedMode webpack:///./app/util/legUtils.js?:512
    StreetModeSelectorButton webpack:///./app/component/StreetModeSelectorButton.js?:59
    React 11
    unstable_runWithPriority webpack:///./node_modules/scheduler/cjs/scheduler.development.js?:653
    React 6
    queryAdditionalItineraries webpack:///./app/component/ItineraryPage.js?:453
[5f6a84d5c3494fabb042.js line 59 > eval:512:41](http://localhost:8080/proxy/js/5f6a84d5c3494fabb042.js%20line%2059%20%3E%20eval)
    getExtendedMode webpack:///./app/util/legUtils.js?:512
</details>

vesameskanen committed:
Merge branch 'v3' into DT-6175-3

vesameskanen committed:
fix: do not assume that first parkride itinerary has public transit legs

vesameskanen committed:
fix: always pass correclty typed carPlan prop

vesameskanen committed:
fix: incorrect message id

vesameskanen committed:
fix: incorrect bicycle leg filtering

vesameskanen committed:
fix: move the rest of state chages away from render method

Other improvements:
- more renaming to make things understandable
- reduce use of unnecessary duplicate state variables

vesameskanen committed:
chore: consistent style file naming

vesameskanen committed:
chore: consistent naming for component classes

vesameskanen committed:
chore: remove dead component

vesameskanen committed:
chore: consistent naming of component class files

vesameskanen committed:
fix: refactor more

- filter relaxed plan after query, not when accessing it
- remove silly complex looking 'nop' functions
- use bettern names

vesameskanen committed:
fix: refactor buggy itinerary selection getter util, rename badly named variables

vesameskanen committed:
fix: refactor getTopics util func

vesameskanen committed:
fix: refactor chaotic itinerary selection code

vesameskanen committed:
fix: co2 setting comes from config only

vesameskanen committed:
fix: correct check for showing car itineraries

vesameskanen committed:
fix: refactor and simplify

- better names
- remove redundant complicated props
- simplify

vesameskanen committed:
fix: mitigate partly the confusion between selectedPlan and combinedItineraries

vesameskanen committed:
chore: tune notifier for showing relaxed itineraries

vesameskanen committed:
fix: define props for mobile and desktop views only once

vesameskanen committed:
fix: remove fuzzy 'walking fastest' condition

vesameskanen committed:
chore: rename to avoid confusion

vesameskanen committed:
fix: move constant prop to child component

vesameskanen committed:
fix: simplify load spinner management and rendering

vesameskanen committed:
chore: simplify more, do not allow earlier/later on parkride page

vesameskanen committed:
fix: remove duplicate mqtt subscribe at component mount

Prop change processing seems to check topics and subscribe if necessary

vesameskanen committed:
fix: remove use of deprecated react method

vesameskanen committed:
fix: rationalize

- remove nonsense check which compares config change
- sensible loading state management
- rename and simplify

vesameskanen committed:
fix: improve querying state management

vesameskanen committed:
fix: refactor settings change management

vesameskanen committed:
fix: path management

vesameskanen committed:
chore: refactor plan selection in render method

vesameskanen committed:
fix: refactor prop change processing

vesameskanen committed:
test: remove test for deleted func

vesameskanen committed:
fix: found one more compoenent without key prop

vesameskanen committed:
fix: some new bugs from recent changes

Itinerary page has 50% less code than before, and basic functionality works.
Error messages and notifiers still need refactoring.

vesameskanen committed:
fix: remove render condition which exceeds human comprehension

optionsome commented on app/component/ItineraryList/ItineraryList.js:
```diff
@@ -140,89 +118,62 @@ function ItineraryList(
+            </div>
+          </div>
+        )}
+        {loadingMore === 'top' && (
```
Would be nice to explicitly know the possible values. We could use enum like solution for this.

vesameskanen replied:
done

optionsome commented on app/component/ItineraryList/ItineraryList.js:
```diff
@@ -262,11 +213,42 @@ function ItineraryList(
+                }
+              />
+            </div>
+          </div>{' '}
```
What is this empty space for?

vesameskanen replied:
Apparently for nothing. Found and fixed plenty of those.

optionsome commented on app/component/ItineraryListContainer.js:
```diff
@@ -203,6 +199,13 @@ class ItineraryListContainer extends React.Component {
   }
 
+  renderMoreButton(arriveBy, position) {
+    if (position === 'top') {
```
Same as above, would be nice to have some enum object that contains possible values and compare against that.

vesameskanen replied:
This jus a local function call parameter. Changed the parameter to boolean.

optionsome commented on app/component/ItineraryPage.js:
```diff
@@ -87,7 +79,21 @@ import ItineraryShape from '../prop-types/ItineraryShape';
 import RoutingErrorShape from '../prop-types/RoutingErrorShape';
 
-const POINT_FOCUS_ZOOM = 16; // used when focusing to a point
+const streetHashes = ['walk', 'bike', 'bikeAndVehicle', 'car', 'parkAndRide'];
```
Not sure if this could belong to path.js or something and maybe again use some enum like thing so we can compare against it when we check what the hash is.

vesameskanen replied:
Fixed.  Consistent  sharing of string constants is not used yet, but maybe it is good time to get started.

optionsome commented on app/component/ItineraryPage.js:
```diff
@@ -314,29 +244,66 @@ class ItineraryPage extends React.Component {
+          result.bikeAndPublicPlan?.itineraries,
+        );
+
+        let n1 = bikeParkItineraries.length;
+        let n2 = bikePublicItineraries.length;
+        if (n1 < 3) {
+          n2 = Math.min(6 - n1, n2);
+        } else if (n2 < 3) {
+          n1 = Math.min(6 - n2, n1);
+        } else {
+          n1 = 3;
+          n2 = 3;
+        }
```
We could add a comment for what this is used for.

vesameskanen replied:
done

optionsome commented on app/component/ItineraryPage.js:
```diff
@@ -485,8 +425,6 @@ class ItineraryPage extends React.Component {
       name: null,
     });
-    this.setState({ loadingMoreItineraries: reversed ? 'bottom' : 'top' });
-    this.showScreenreaderLoadingAlert();
```
A bit worried what removing this means.

optionsome commented on app/component/StreetModeSelectorButton.js:
```diff
@@ -61,35 +55,19 @@ export const StreetModeSelectorButton = (
-    if (mode === 'subway') {
-      metroColor = '#CA4000';
-    }
+    secondaryColor = mode === 'subway' ? '#CA4000' : '';
```
Hardcoded subway color feels a bit wrong.

vesameskanen replied:
Not  only a bit. Fixed.

vesameskanen committed:
feat: use shared costants for spinner position prop

vesameskanen committed:
fix: use boolean instead of string to define button position

vesameskanen committed:
fix: erase strange redundant empty space blocks from rendering

vesameskanen committed:
chore: explain the logic how bike + transit itnerary list is assembled

vesameskanen committed:
fix: no hardcoded colors

vesameskanen committed:
feat: use shared constants for itinerary page hash management

optionsome commented on app/component/StreetModeSelectorButton.js:
```diff
@@ -31,14 +32,14 @@ export const StreetModeSelectorButton = (
   let distance = 0;
   switch (name) {
-    case 'WALK':
+    case streetHash.walk:
```
Why was it 'WALK'? Does this work?

vesameskanen committed:
fix: driving distance in park&ride button
